client/widgets/multi_slider.cpp

```diff
@@ -0,0 +1,630 @@
+// SPDX-License-Identifier: GPLv3-or-later
+// SPDX-FileCopyrightText: Louis Moureaux <m_louis30@yahoo.com>
+
+#include "widgets/multi_slider.h"
+
+#include "log.h"
+
+#include <QFocusEvent>
+#include <QKeyEvent>
+#include <QMouseEvent>
+#include <QPainter>
+
+#include <algorithm>
+#include <cmath>
+
+namespace {
+/// Colors
+namespace colors {
+/// Focus indicator (thin line under one of the categories)
+const QColor focus_indicator = Qt::gray;
+/// Background of the handles
+const QColor handle_background = Qt::lightGray;
+/// Small dot on the handles
+const QColor handle_indicator = Qt::gray;
+/// Small dot on the handles, when the handle is hovered
+const QColor handle_hover = Qt::darkGray;
+/// Small dot on the handles, when the handle is being dragged
+const QColor handle_dragged = handle_hover;
+} // namespace colors
+/// Widget dimensions, in logical pixels
+namespace metrics {
+/// Gap between the icons and the focus indicator
+const double focus_bar_gap = 3;
+/// Height of the focus indicator
+const double focus_bar_height = 1;
+/// Gap at the top of the handle bar
+const double handle_bar_gap = 2;
+/// Width of the handle bar
+const double handle_bar_width = 4;
+/// Gap between the icons and the handle (circle)
+const double handle_gap = 1;
+/// Radius of the handle
+const double handle_radius = 8;
+/// Radius of the small disk inside the handle
+const double handle_indicator_radius = 4;
+/// Radius of the small disk inside the handle, when the handle is active
+const double handle_active_indicator_radius = handle_indicator_radius + 1;
+/// Height added to the icon height by control elements (handle etc)
+const double extra_height = std::max(focus_bar_gap + focus_bar_height,
+                                     handle_gap + handle_radius * 2);
+} // namespace metrics
+} // anonymous namespace
+
+namespace freeciv {
```

Comment:
When we refactor something why don't we use a diff namespace such as fc21 or freeciv21?

Reply:
The `namespace freeciv` is actually new. Could switch to `fc21` but I find it a bit obscure and it's also harder to type for me (I need shift to type numbers). `freeciv21` is probably too long.

Reply:
Ok learned something new.